packages/legend-query/src/components/QueryBuilderParameterPanel.tsx

```diff
@@ -389,6 +405,175 @@ const VariableExpressionViewer = observer(
   },
 );
 
+const EditMilestoningParemter = observer(
+  (props: { queryBuilderState: QueryBuilderState; index: number }) => {
+    const { queryBuilderState, index } = props;
+    const handleDrop = useCallback(
+      (item: QueryBuilderParameterDragSource): void => {
+        queryBuilderState.querySetupState.classMilestoningValue[index] =
+          item.variable.parameter;
+      },
+      [queryBuilderState, index],
+    );
+    const [{ isFilterValueDragOver }, dropConnector] = useDrop(
+      () => ({
+        accept: [QUERY_BUILDER_PARAMETER_TREE_DND_TYPE.VARIABLE],
+        drop: (
+          item: QueryBuilderParameterDragSource,
+          monitor: DropTargetMonitor,
+        ): void => {
+          if (!monitor.didDrop()) {
+            handleDrop(item);
+          }
+        },
+        collect: (monitor): { isFilterValueDragOver: boolean } => ({
+          isFilterValueDragOver: monitor.isOver({ shallow: true }),
+        }),
+      }),
+      [handleDrop],
+    );
+    const resetMilestoningParameter = (): void => {
+      queryBuilderState.querySetupState.classMilestoningValue[index] =
+        new PrimitiveInstanceValue(
+          GenericTypeExplicitReference.create(
+            new GenericType(
+              queryBuilderState.graphManagerState.graph.getPrimitiveType(
+                PRIMITIVE_TYPE.LATESTDATE,
+              ),
+            ),
+          ),
+          queryBuilderState.graphManagerState.graph.getTypicalMultiplicity(
+            TYPICAL_MULTIPLICITY_TYPE.ONE,
+          ),
+        );
+    };
+    return (
+      <div
+        ref={dropConnector}
+        className="query-builder-filter-tree__condition-node__value dnd__overlay__container"
```

Comment:
why do you bring `filter-tree` stuff over here? Be mindful when copy/paste

Reply:
Actually `MilestoningParameterEditor` uses similar logic as `filter-tree` for drag and drop and editing variables. My bad forgot to rename the stuff here